src/engine/BMInterface.php

```diff
@@ -1086,27 +1056,31 @@ protected function resolve_random_button_selection(BMGame &$game) {
     }
 
     protected function does_need_random_button_selection(BMGame $game) {
+        $hasRandomlyChosenButtons = FALSE;
+        $hasUnresolvedNames = FALSE;
+        foreach ($game->playerArray as $player) {
+            // do not resolve random names unless all players have joined the game
```

Comment:
Nit: "do not resolve random *button* names".  ("What's a random player name?" i asked myself on first read.)